### ISSUE TITLE
Return HTML from hooks to self-replace.

### ISSUE THREAD

**Issue by romaninsh** [atk4/ui#214]
``` php
$b = new Button();
$app->add($b);

$b->onClick(function() use($b) {
    return $b->addClass('red');
});
```

When button is clicked it should replace itself with new provided HTML.

**Comment by DarkSide666** [comment]
`return $b->addClass('red');` is not HTML, but object.
Not sure yet if this will be good or bad. Why self-replacing is something different that doing any other action on other views in hooks?

**Comment by romaninsh** [comment]
You could replace form with message.

**Comment by DarkSide666** [comment]
Isn't something like this already working?

```
$b = new Button();
$f = new Form();
$app->add($b);
$app->add($f);

$b->onClick(function($b) use ($f) {
    $b->addClass('red')->set('Thanks for clicking'); // change button
    $f->destroy(); // no more form at all
});
```

but, yeah, i guess you're right - we probably need to return HTML code (or renderable object) and that could replace object on which hook is called. In this case - button. Otherwise we have to use JS chain to make some changes in layout.

**Comment by romaninsh** [comment]
partially resolved in #179
